hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -77,6 +77,10 @@ public HiveSyncTool(HiveSyncConfig cfg, HiveConf configuration, FileSystem fs) {
     super(configuration.getAllProperties(), fs);
 
     try {
+      if (cfg.useKerberos) {
+        configuration.set("hive.metastore.sasl.enabled", "true");
+        configuration.set("hive.metastore.kerberos.principal", cfg.kerberosPrincipal);
```

Comment:
Does this work for both `hms` and `jdbc` sync mode ?

Reply:
The error is due to HoodieHiveClient call to method Hive.get(configuration).getMSC() , whether JDBC or HMS or HIVEQL calls this method,Therefore, So ALL I have to do is solve the problem of Hive.get(configuration).getMSC() to solve all the connection problems,I added a parameter to solve this problem

Reply:
Both HMS and JDBC can use this parameter to connect to Kerberos-managed Hive

Reply:
But only if the task execution node has authenticated kerberos identity

Reply:
Application startup is already kerberos authentication, setting two parameters has no effect, call Hive.get(configuration).getMSC() still failed

Reply:
If you call method Hive.get(configuration).getMSC() in Hive.get(configuration).getMSC()  test class, you need to pass it three arguments
conf.set("metastore.thrift.uris", "thrift://ip:host");
conf.set("hive.metastore.kerberos.principal", "hive_meta_principla");
conf.set("hive.metastore.sasl.enabled", "true");

Reply:
You can test kerberos in this way  
Method KerberosLogin.login() performs Kerberos authentication on the client
  public static void test() {
        KerberosLogin.login();
        HiveConf conf = new HiveConf();
        conf.set("metastore.thrift.uris", "thrift://ip:9083");
        conf.set("hive.metastore.kerberos.principal", "xxxxxxxxx");
        conf.set("hive.metastore.sasl.enabled", "true");
        try {
            IMetaStoreClient msc = Hive.get(conf).getMSC();
            boolean b = msc.tableExists("test", "a_message_info_test");
            System.out.println(b);
        } catch (MetaException e) {
            e.printStackTrace();
        } catch (HiveException e) {
            e.printStackTrace();
        } catch (UnknownDBException e) {
            e.printStackTrace();
        } catch (TException e) {
            e.printStackTrace();
        }
    }

Reply:
If the Hive dependent version used is different from the dependent version used to access Hive, the connection may fail

Reply:
https://github.com/melin/hudi/commit/796c8546f81f5a5cb22ff62462d4ee48623cc15b
Try to create a hiveClient,hive-site. in the ShowPartitionsHoodieCommand The xml contains these configuration parameters, and the initial hiveClient failed

val hadoopConf = sparkSession.sessionState.newHadoopConf()
val hiveConf = new HiveConf()
val hiveClient = new HoodieHiveClient(cfg, hiveConf, fs)

![image](https://user-images.githubusercontent.com/1145830/132623175-761fc222-e297-45da-bfdf-43882a0fa46c.png)

Reply:
![image](https://user-images.githubusercontent.com/90183930/132626141-ddec15ed-66c0-4ce8-8b1f-83faa166a3e4.png)
![Uploading image.png…]()

Reply:
KerberosLogin.login();
        HiveConf hiveConf = new HiveConf();
        HiveSyncConfig cfg = new HiveSyncConfig();
        FileSystem fs = FileSystem.newInstance(new Configuration());

        hiveConf.set("metastore.thrift.uris", "thrift://ip:9083");
        hiveConf.set("hive.metastore.kerberos.principal", "hive/_HOST@BIGDATA");
        hiveConf.set("hive.metastore.sasl.enabled", "true");

        cfg.basePath = "hdfs://ip:8020/warehouse/hudi/t19";
        cfg.syncMode = "HMS";
        cfg.databaseName = "test";

        HoodieHiveClient hoodieHiveClient = new HoodieHiveClient(cfg, hiveConf, fs);
        hoodieHiveClient.createDatabase("test_kerbero_table");

Reply:
This is the test I did just now, there is no problem, you can report the error you encounter to me

Reply:
![image](https://user-images.githubusercontent.com/90183930/132626418-87372793-cd3c-4516-bc0f-0615765a099b.png)

Reply:
![image](https://user-images.githubusercontent.com/90183930/132627364-14c9eeec-99fd-48b6-a86f-8be4b223e905.png)
![image](https://user-images.githubusercontent.com/90183930/132627405-95d86c56-bb9d-4b5f-9d9c-7770ab175192.png)
Even if I provided the hive meta uri information, it would not affect me to create A Kerberos-managed Hive through hoodieHiveClient

Reply:
I just tested that even if I provided the hive-site.xml , it would not affect me to create A Kerberos-managed Hive through hoodieHiveClient
![image](https://user-images.githubusercontent.com/90183930/132627637-dbcd8e00-475f-48ce-b019-a3ca8fa553af.png)